Lib/typing.py

```diff
@@ -1089,11 +1120,15 @@ def _typevartuple_prepare_subst(self, alias, args):
     elif left + right > alen:
         raise TypeError(f"Too few arguments for {alias};"
                         f" actual {alen}, expected at least {plen-1}")
+    if left == alen - right and self.__default__ is not None:
+        replacement = _unpack_args([self.__default__])
+    else:
+        replacement = args[left: alen - right]
```

Comment:
blegh, maybe I'll do a PR going through the whole file replacing all the `alen` and `elen` variables after this is merged 😛

Reply:
Thoughts on making `_unpack_args` accept a variadic number of arguments? Constructing a list here just to pass it to `_unpack_args` feels kinda strange. Something like this?

```diff
--- a/Lib/typing.py
+++ b/Lib/typing.py
@@ -331,7 +331,7 @@ def _check_generic_specialization(cls, arguments):
                         f" for {cls}; actual {actual_len}, expected {expect_val}")
 
 
-def _unpack_args(args):
+def _unpack_args(*args):
     newargs = []
     for arg in args:
         subargs = getattr(arg, '__typing_unpacked_tuple_args__', None)
@@ -1121,7 +1121,7 @@ def _typevartuple_prepare_subst(self, alias, args):
         raise TypeError(f"Too few arguments for {alias};"
                         f" actual {alen}, expected at least {plen-1}")
     if left == alen - right and self.__default__ is not None:
-        replacement = _unpack_args([self.__default__])
+        replacement = _unpack_args(self.__default__)
     else:
         replacement = args[left: alen - right]
 
@@ -1427,8 +1427,7 @@ def __getitem__(self, args):
         # Preprocess `args`.
         if not isinstance(args, tuple):
             args = (args,)
-        args = tuple(_type_convert(p) for p in args)
-        args = _unpack_args(args)
+        args = _unpack_args(*(_type_convert(p) for p in args))
         new_args = self._determine_new_args(args)
         r = self.copy_with(new_args)
         return r
         ```

Reply:
Sure, done.